lib/src/rules/prefer_constructors_over_static_methods.dart

```diff
@@ -78,10 +78,12 @@ class _Visitor extends SimpleAstVisitor<void> {
   void visitMethodDeclaration(MethodDeclaration node) {
     final returnType = node.returnType?.type;
     final parent = node.parent;
+    final typeParameters = node.typeParameters?.typeParameters ?? [];
     if (node.isStatic &&
         parent is ClassOrMixinDeclaration &&
         returnType is InterfaceType &&
-        returnType.element == parent.declaredElement &&
+        returnType == parent.declaredElement.thisType &&
+        typeParameters.isEmpty &&
```

Comment:
Rather than create an empty list for every non-generic method, consider dropping line 81 and using `node.typeParameters != null` here instead. (I don't think we need to guard against a malformed declaration in which there is an empty list of type parameters.)

Reply:
Haha definitely simpler. Thanks